src/Cli/Microsoft.DotNet.Cli.Utils/FrameworkDependencyFile.cs

```diff
@@ -111,7 +109,7 @@ internal static bool TryGetMostFitRuntimeIdentifier(
 
         private DependencyContext CreateDependencyContext()
         {
-            using (Stream depsFileStream = File.OpenRead(_depsFilePath))
+            using (Stream depsFileStream = File.OpenRead(_depsFilePath ?? string.Empty))
```

Comment:
I didn't opt to adding any additional logic, especially throwing exceptions, as I originally didn't intend on changing ANY logic. However, we decided that we didn't want to use the null-forgiving operator (!), so the only changes I've made in logic is checking for null and letting the program "continue" if it was null, as you saw above in your one comment.

Reply:
Before this, it would've failed (with a NRE) if _depsFilePath were somehow null, so this actually does change the logic to make it fail in a different way. I think "_depsFilePath is unexpectedly null" is more clear and accurate than "could not find file ''", which might have multiple causes.

I commented on multiple like this where I think the error you end up getting from ?? string.Empty (or equivalent) is less clear and actionable than what the error otherwise would've been. I think there's a lot of potential benefit from nullable (and hence from a PR like this), but I also think it's important to not miss the woods for the trees: bringing an error forward so we can act on it is ideal. Hiding it to make an analyzer happy is not. Though I agree I'm proposing adding more than you did, I think it would make it clearer rather than less clear. Does that make sense?

Reply:
Yes, @marcpopMSFT is aware the logic will change in unintended ways, but we have not opted on redesigning the code around nullability, as that is a major project. We're only making the code handle nullability support. If that support results in bugs, that is an understood risk and not in the scope of this PR.

Your concerns are **exactly** the concerns I brought up with Marc, so I'm on your side for this. But I'm not supposed to invest the effort in making things better, only making them compatible. It would take **way too long** to write this code properly to handle nullability in the design of it. There are *thousands* of situations just like this and we don't have a way to make them less error prone without knowing the entire codebase. As it stands, I'm doing the minimum viable changes to make the code compatible with nullability, which would still take *a while*. It is not intended to make the code *better*. Changing a potential null-ref error into a different error is **completely reasonable** for this PR.

Reply:
The guidance I got from other teams was to make the switch over first and then start investing in changing hte logic. They did indicate that changing the logic to provide better error handling and messaging was a desired outcome but this switch is already going to be quite challenging. I wonder if it makes sense to create an issue with a list of all the spots we want to revisit later?

Reply:
I don't know if it is possible to create that list as it would be a majority of the changes. Most uses of `string` in the codebase fall into this category, and there are likely *thousands* of strings.